python.d/ceph.chart.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+# Description: ceph netdata python.d module
+# Author: Luis Eduardo (lets00)
+
+try:
+    import rados
+    CEPH = True
+except ImportError:
+    CEPH = False
+
+import json
+from bases.FrameworkServices.SimpleService import SimpleService
+
+# default module values (can be overridden per job in `config`)
+update_every = 10
+priority = 60000
+retries = 60
+
+ORDER = ['general', 'pool_usage', 'pool_objects', 'osd_usage']
+
+CHARTS = {
+    'general': {
+        'options': [None, 'Ceph General Options', 'KB', 'general', 'ceph.general', 'stacked'],
+        'lines': [
+            ['general_available', 'avail', 'absolute', 1, 1024],
+            ['general_usage', 'used', 'absolute', 1, 1024]
+        ]
+    },
+    'pool_usage': {
+        'options': [None, 'Ceph Pools', 'KB', 'pool', 'ceph.pool_usage', 'line'],
+        'lines': []
+    },
+    'pool_objects': {
+        'options': [None, 'Ceph Pools', 'objects', 'pool', 'ceph.pool_objects', 'line'],
+        'lines': []
+    },
+    'osd_usage': {
+        'options': [None, 'Ceph OSDs', 'KB', 'osd', 'ceph.osd_usage', 'line'],
+        'lines': []
+    }
+}
+
+
+class Service(SimpleService):
+    def __init__(self, configuration=None, name=None):
+        SimpleService.__init__(self, configuration=configuration, name=name)
+        self.order = ORDER
+        self.definitions = CHARTS
+        self.config_file = self.configuration.get('config_file')
+        self.keyring_file = self.configuration.get('keyring_file')
+        self.cluster = rados.Rados(conffile=self.config_file,
+                                   conf=dict(keyring=self.keyring_file))
+
+    def check(self):
+        """
+        Checks module
+        :return:
+        """
+        if not CEPH:
+            self.error('rados module is needed to use ceph.chart.py')
+            return False
+        if not (self.config_file and self.keyring_file):
+            self.error('config_file and/or keyring_file is not defined')
+            return False
+        try:
+            self.cluster.connect()
+        except Exception as error:
```

Comment:
I think `self.cluster.connect()` should cause only rados related exceptions (d) so should be
```python
 except rados.Error as error
```

Reply:
Makes sense. I will change this to treat only rados exception.